packages/voting-contract/src/lib.rs

```diff
@@ -197,18 +198,24 @@ where
 {
     // anyone can trigger this if the vote passed
 
-    let mut prop = proposals::<P>().load(deps.storage, proposal_id)?;
-    if [Status::Executed, Status::Rejected, Status::Passed]
+    let mut prop = proposals().load(deps.storage, proposal_id)?;
+
+    if prop.status == Status::Rejected {
+        return Err(ContractError::Rejected {});
+    }
+
+    prop.update_status(&env.block);
+
+    if [Status::Executed, Status::Passed]
         .iter()
         .any(|x| *x == prop.status)
     {
         return Err(ContractError::WrongCloseStatus {});
     }
-    if !prop.expires.is_expired(&env.block) {
+    if prop.status == Status::Open {
```

Comment:
I don't think this is right. This will allow closing `Pending` proposals that are not expired yet.

Reply:
Okay, I reverted that change.
It's redundant to check expiration again since update status does that as well, but otherwise I'd have to create another 3 line iter with statuses `Pending` and `Open` as above, or just long if condition, which is tedious... and result is the same in the end.